mastodon/src/main/java/org/joinmastodon/android/fragments/ComposeFragment.java

```diff
@@ -919,14 +919,18 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater){
 
 
 		languageButton = wrap.findViewById(R.id.language_btn);
-		languageButton.setOnClickListener(v->showLanguageAlert());
-		languageButton.setOnLongClickListener(v->{
-			if(!getLocalPrefs().bottomEncoding){
-				getLocalPrefs().bottomEncoding=true;
-				getLocalPrefs().save();
-			}
-			return false;
-		});
+		if(instance.isIceshrimpJs())
+			languageButton.setVisibility(View.GONE);
+		else {
+			languageButton.setOnClickListener(v->showLanguageAlert());
+			languageButton.setOnLongClickListener(v->{
+				if(!getLocalPrefs().bottomEncoding){
+					getLocalPrefs().bottomEncoding=true;
+					getLocalPrefs().save();
+				}
+				return false;
+			});
+		}
```

Comment:
It should be enough to just hide the button, to avoid the additional code indentation.